### PR TITLE
Ensure backwards compatibility for legacy H5 saving format

### PR DESCRIPTION
**Pull request by nkovela1**
This PR ensures backwards compatibility for the legacy H5 saving format, particularly paying attention to differences in configs, arguments, and deserialization methods between tf.keras and Keras Core.

A few tests have been added showing a tf.keras model saved into H5 and then loaded into Keras Core to become a Keras Core model.
The test suite will be expanded in subsequent PRs to ensure greater coverage.

### CODE REVIEW

File: keras_core/legacy/saving/legacy_h5_format_test.py [hunk 1]
```diff
@@ -121,3 +121,45 @@ def test_compiled_model_with_various_layers(self):
...
+        self._check_reloading_model(ref_input, model, tf_keras_model)
+
+    def test_compiled_model_with_various_layers(self):
+        model = models.Sequential()
+        model.add(layers.Dense(2, input_shape=(3,)))
+        model.add(layers.RepeatVector(3))
+        model.add(layers.TimeDistributed(layers.Dense(3)))
+        model.compile(optimizer="rmsprop", loss="mse")
+
+        tf_keras_model = tf.keras.Sequential()
```
Comment (fchollet) [inline keras_core/legacy/saving/legacy_h5_format_test.py:156]: In the future `tf.keras` will be Keras Core. So can we avoid depending on it, and instead depend on a static artifact? Maybe either downloaded or checked into git?

File: keras_core/legacy/saving/serialization.py [hunk 2]
```diff
@@ -484,6 +484,12 @@ def deserialize(config, custom_objects=None):
             arg_spec = inspect.getfullargspec(cls.from_config)
             custom_objects = custom_objects or {}
 
+            # TODO(nkovela): Replace during Keras 3.0 release
+            # Replace keras refs with keras_core
```
Comment (fchollet) [inline keras_core/legacy/saving/serialization.py:488]: After the swap, the replacement should go the other way

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_core/legacy/saving/legacy_h5_format_test.py (hunk 1 above)
Comment (nkovela1) [reply keras_core/legacy/saving/legacy_h5_format_test.py:156]: Agreed, however, for now as I develop and expand this testing suite I will keep this dependency for debugging purposes/changes. After the full testing suite is done (in the next PR or so), I will generate them as h5 artifacts and add them to that PR.

File: keras_core/legacy/saving/serialization.py (hunk 2 above)
Comment (nkovela1) [reply keras_core/legacy/saving/serialization.py:488]: Makes sense, fixed the TODO comment.

**Pull request merged**